src/PIL/PpmImagePlugin.py

```diff
@@ -49,26 +49,52 @@ class PpmImageFile(ImageFile.ImageFile):
     format = "PPM"
     format_description = "Pbmplus image"
 
-    def _token(self, s=b""):
+    def _read_magic(self, magic=b""):
         while True:  # read until next whitespace
             c = self.fp.read(1)
-            if not c or c in b_whitespace:
+            if c in B_WHITESPACE:
```

Comment:
You've removed the `not c` condition. The purpose of that would have been to avoid an infinite loop if `c` was empty.
i mean, technically, an empty string is in `B_WHITESPACE`, so it will pass, but your version seems less clear.

Reply:
Since the [Library Reference](https://docs.python.org/3/reference/expressions.html#in) states that:
> Empty strings are always considered to be a substring of any other string, so "" in "abc" will return `True`.

I considered the check to be redundant. Maybe a short comment like `# also checks for empty string` would make it clearer? Change it as you see fit.

Reply:
I've pushed a commit to use `not c`.